### PR TITLE
Extend to add Default Server level configs for Dedup Tables

### PR DESCRIPTION
**Pull request by deepthi912**
The following config settings are added in the PR on server level to extend the default behaviour of dedup:
1) EnablePreload
2) metadataManagerClass

### CODE REVIEW

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java [hunk 1]
```diff
@@ -96,4 +96,10 @@ public String getDedupTimeColumn() {
   public boolean isEnablePreload() {
     return _enablePreload;
   }
+
+  // CHECKSTYLE:OFF
```
Comment (deepthi912) [inline pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java:100]: Switched off the checkstyle as we would need to override server configs when extending

File: pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManagerConfig.java [hunk 2]
```diff
@@ -49,6 +49,8 @@ public class HelixInstanceDataManagerConfig implements InstanceDataManagerConfig
   public static final String SEGMENT_DIRECTORY_LOADER = "segment.directory.loader";
   // Prefix for upsert config
   public static final String UPSERT_CONFIG_PREFIX = "upsert";
+  // Prefix for upsert config
```
Comment (klsince) [inline pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManagerConfig.java:52]: typo: ‘for dedup’

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java:100]: This shouldn't be needed. Do you run into error without it?

Comment (deepthi912) [reply pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java:100]: Yeah ran into checkstyle exception, like setter for enablePreload and enableSnapshot cannot be set in dedupConfig

Comment (Jackie-Jiang) [reply pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java:100]: I know why it doesn't like it. This method should not return anything. Change the return to `void` should be good

Comment (deepthi912) [reply pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java]: My bad, changed.

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**